examples/demo-rollup/src/test_rpc.rs

```diff
@@ -0,0 +1,205 @@
+        
+        query_test_helper(data, expected);
+
+        let data = r#"{"jsonrpc":"2.0","method":"ledger_getTransactions","params":[[{ "batch_id": 1, "offset": 0}]],"id":1}"#
+            .as_bytes();
+        let expected = r#"{"jsonrpc":"2.0","result":[{"hash":"0x329c58b0a9b08973bed32452c2cefa0ab567146505711337c955b24cf41c6e99","event_range":{"start":1,"end":1},"body":[116,120,49,32,98,111,100,121],"custom_receipt":0}],"id":1}"#;
+        query_test_helper(data, expected);
+
+        let data = r#"{"jsonrpc":"2.0","method":"ledger_getTransactions","params":[[{ "batch_id": 1, "offset": 1}]],"id":1}"#
+            .as_bytes();
+        let expected = r#"{"jsonrpc":"2.0","result":[{"hash":"0xae87c50974dc43f4f70e84cb27e4630e4e47f782cff1e3d484310d82cea9acf6","event_range":{"start":1,"end":3},"body":[116,120,50,32,98,111,100,121],"custom_receipt":1}],"id":1}"#;
+        query_test_helper(data, expected);
+        
+        
+        let data = r#"{"jsonrpc":"2.0","method":"ledger_getBatches","params":[[2], "Standard"],"id":1}"#
+            .as_bytes();
+        let expected = r#"{"jsonrpc":"2.0","result":[{"hash":"0x41c8790eb95a24d1b4aabc606e88c602073c72ada51ebc72300a82591dc49459","tx_range":{"start":3,"end":4},"txs":["0x329c58b0a9b08973bed32452c2cefa0ab567146505711337c955b24cf41c6e99"],"custom_receipt":1}],"id":1}"#;
+        query_test_helper(data, expected);
+
+        let data = r#"{"jsonrpc":"2.0","method":"ledger_getBatches","params":[[1], "Compact"],"id":1}"#
+            .as_bytes();
+        let expected = r#"{"jsonrpc":"2.0","result":[{"hash":"0xf344c1da53b56f7a49d02ee20899fb914bdfc3632db4f337ade196e72f5eb083","tx_range":{"start":1,"end":3},"custom_receipt":0}],"id":1}"#;
+        query_test_helper(data, expected);
+
+        let data = r#"{"jsonrpc":"2.0","method":"ledger_getBatches","params":[[0], "Compact"],"id":1}"#
+            .as_bytes();
+        let expected = r#"{"jsonrpc":"2.0","result":[null],"id":1}"#;
+        query_test_helper(data, expected);
+
+        let data = r#"{"jsonrpc":"2.0","method":"ledger_getEvents","params":[1],"id":1}"#.as_bytes();
+        let expected = r#"{"jsonrpc":"2.0","result":[{"key":[101,118,101,110,116,49,95,107,101,121],"value":[101,118,101,110,116,49,95,118,97,108,117,101]}],"id":1}"#;
+        query_test_helper(data, expected);
+
+        let data = r#"{"jsonrpc":"2.0","method":"ledger_getEvents","params":[2],"id":1}"#.as_bytes();
+        let expected = r#"{"jsonrpc":"2.0","result":[{"key":[101,118,101,110,116,50,95,107,101,121],"value":[101,118,101,110,116,50,95,118,97,108,117,101]}],"id":1}"#;
+        query_test_helper(data, expected);
+
+        let data = r#"{"jsonrpc":"2.0","method":"ledger_getEvents","params":[3],"id":1}"#.as_bytes();
+        let expected = r#"{"jsonrpc":"2.0","result":[null],"id":1}"#;
+        query_test_helper(data, expected);
```

Comment:
nit: It's usually considered good style to have separate `#[test]`s for each logical unit of code - in this case, each RPC method. That way, you get a nice printout showing exactly what's broken when the test fails instead of having to do extra analysis to figure out which endpoint is broken. 

It will cost us a little bit of efficiency, since we'll have to set up and tear down the tokio::runtime with each test, but these tests are still so lightweight that it shouldn't matter.

Reply:
I divided this test into 4 subtests: one for each rpc function